pkgs/development/python-modules/unrar2-cffi/default.nix

```diff
@@ -0,0 +1,46 @@
+{
+  buildPythonPackage,
+  cffi,
+  fetchPypi,
+  lib,
+  nix-update-script,
+  pythonOlder,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "unrar2-cffi";
+  version = "0.4.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "unrar2_cffi";
+    hash = "sha256-z8jOntg/d9c/ogtgkum5AXt7oKCFYj8ggvQNTZtp724=";
+  };
+
+  build-system = [
+    cffi
+    setuptools
+  ];
+
+  dependencies = [
+    cffi
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Read RAR file from python -- cffi edition";
+    homepage = "https://github.com/noaione/unrar2-cffi";
+    changelog = "https://github.com/noaione/unrar2-cffi/releases/tag/v${version}";
+    license = with lib; [
+      licenses.asl20
+      licenses.unfreeRedistributable # for including unrar
```

Comment:
Is unrar vendored? I don't see it in the package definition.

Reply:
Yeah, it's built by unrar2-cffi to support the FFI. I believe the unrar source is shipped as part of the pypi package.